Description: Backport of upstream commit bd7b3d6d47ccbb5e1747126f8e2a297f38e56b8c
Author: Emanuele Rocca <ema@wikimedia.org>
Last-Update: 2020-04-09

commit bd7b3d6d47ccbb5e1747126f8e2a297f38e56b8c
Author: Martin Blix Grydeland <martin@varnish-software.com>
Date:   Tue Oct 1 11:17:17 2019 +0200

    Clear err_code and err_reason at start of request handling
    
    req->err_code and req->err_reason are set when going to synthetic
    handling. From there the resp.reason HTTP field is set from
    req->err_reason if set, or the generic code based on req->err_code is used
    if it was NULL. This patch clears these members so that a value from the
    handling of a previous request doesn't linger.
    
    Fixes: VSV00004

diff --git a/bin/varnishd/cache/cache_req_fsm.c b/bin/varnishd/cache/cache_req_fsm.c
index cdd443fad..1c18bf715 100644
--- a/bin/varnishd/cache/cache_req_fsm.c
+++ b/bin/varnishd/cache/cache_req_fsm.c
@@ -797,6 +797,9 @@ cnt_recv(struct worker *wrk, struct req *req)
 	req->client_identity = NULL;
 	req->storage = NULL;
 
+	req->err_code = 0;
+	req->err_reason = NULL;
+
 	http_CollectHdr(req->http, H_Cache_Control);
 
 	if (req->req_body_status == REQ_BODY_FAIL) {
